core/src/main/java/com/yahoo/oak/OakBuffer.java

```diff
@@ -107,4 +107,22 @@
      *                                   minus seven
      */
     double getDouble(int index) throws NullPointerException;
+
+    /*****************************************************************************
+     * CharSequence API
+     *****************************************************************************/
+
+    default int length() {
+        // length in char not in bytes
+        return capacity() / Character.BYTES;
```

Comment:
This is why I dropped the idea of using CharSequence API in PR#172.
`Character.BYTES=2`, so if the capacity is uneven, then the hash ignores the last byte.
I suggest merging PR#172 before this PR, then merging this PR with the new master.
You can avoid rebasing because we squash the commits anyway.
Actually, according to many engineers I discussed with, rebasing it discouraged.

Reply:
OK